test/functional/flat.py

```diff
@@ -0,0 +1,112 @@
+# flat.py
+#
+# A storage for parameters read from a flat (not tested) XML structure.
+#
+# Copyright (c) 2017 vcdMaker team
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+# OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+# IN THE SOFTWARE.
+
+import os
+
+
+class Flat(object):
+    """A storage for parameters read from a flat XML structure."""
+
+    def __init__(self, node, parameters):
+        """Class constructor.
+
+        Arguments:
+        node - The XML node to be read.
+        parameters - The structure describing requirements towards the
+                         parameters to be read.
+
+        Example:
+        parameters =  {'name': ['', ''],
+                       'output_file': ['', 'Missing output file'],
+                       'golden_file': ['', 'Missing golden file']}
+
+        The 'key' is the name of the parameter to be read.
+
+        The first element of the list is the value of the read parameter.
+
+        The second element of the list if empty means that the parameter
+        is optional. If set, it means the parameter is obligatory and it
+        represents a message to be displayed when the parameter is missing.
+        """
+        self.errors = []
+        self.parameters = parameters
+        self.read_parameters(node)
+        self.check_parameters()
+
+    def read_parameters(self, node):
+        """Reads parameters."""
+
+        for element in list(node):
+            if element.tag in self.parameters:
+                self.parameters[element.tag][0] = element.text
+
+    def check_parameters(self):
+        """Checks if all necessary parameters are provided.
+
+        Raises:
+        RuntimeError is raised if not all requested parameters are provided.
+        """
+
+        for element in self.parameters:
+            if self.parameters[element][1] != '':
+                if ((self.parameters[element][0] is None) or
+                        (self.parameters[element][0].strip() == '')):
```

Comment:
For empty string check PEP8 recommends to test agains `False`. So: `if not string:`.